sdk/core/core-rest-pipeline/src/policies/bearerTokenAuthenticationPolicy.ts

```diff
@@ -188,26 +216,96 @@ export function bearerTokenAuthenticationPolicy(
       try {
         response = await next(request);
       } catch (err: any) {
-        error = err;
-        response = err.response;
+        if (isRestError(error)) {
+          response = err.response;
+          error = err;
+        } else {
+          throw err;
+        }
       }
 
-      if (
-        callbacks.authorizeRequestOnChallenge &&
-        response?.status === 401 &&
-        getChallenge(response)
-      ) {
-        // processes challenge
-        const shouldSendRequest = await callbacks.authorizeRequestOnChallenge({
-          scopes: Array.isArray(scopes) ? scopes : [scopes],
-          request,
-          response,
-          getAccessToken,
-          logger,
-        });
-
-        if (shouldSendRequest) {
-          return next(request);
+      async function handleCaeChallenge(claims: string): Promise<PipelineResponse> {
+        const shouldSendRequest = await authorizeRequestOnCaeChallenge(
+          {
+            scopes: Array.isArray(scopes) ? scopes : [scopes],
+            response,
+            request,
+            getAccessToken,
+            logger,
+          },
+          claims,
+        );
+        // Send updated request and handle response for RestError
+        try {
+          if (shouldSendRequest) {
+            error = undefined;
+            response = await next(request);
+          }
+        } catch (err: any) {
+          if (isRestError(error)) {
+            response = err.response;
+            error = err;
+          } else {
+            throw err;
+          }
+        }
+        return response;
+      }
+
+      if (isChallengeResponse(response)) {
+        let claims = getCaeChallengeClaims(response);
+        // Handle CAE by default when receive CAE claim
+        if (claims) {
+          let parsedClaim: string;
+          // Return the response immediately if claims is not a valid base64 encoded string
+          try {
+            parsedClaim = atob(claims);
+          } catch (e) {
+            logger.warning(
+              `The WWW-Authenticate header contains "claims" that cannot be parsed. Unable to perform the Continuous Access Evaluation authentication flow.`,
+            );
+            return response;
```

Comment:
At this point, does the response contain the same error message as the logger or it's just a 401 with whatever message the service returned?

Reply:
The logger should not interfere with the response message the service returns. This catch block is for claims that are not base 64 encoded, and there is a test for this scenario.